### PR TITLE
Robustify SLURM for SIZE==1 and some minor changes

### PR DESCRIPTION
**Pull request by boeddeker**

### CODE REVIEW

File: dlp_mpi/mpi.py [hunk 1]
```diff
@@ -74,6 +87,7 @@
         )
         raise
 
+if not _mpi_available:
```
Comment (thequilo) [inline dlp_mpi/mpi.py:90]: Why this line? Is it for readability?

File: dlp_mpi/util.py [hunk 2]
```diff
@@ -70,3 +70,22 @@ def ensure_single_thread_numeric():
                 'export MKL_NUM_THREADS=1\n\n'
                 f'But you use: {key}={os.environ.get(key)}'
             )
+
+
+def maybe_warn_if_slurm():
+    SLURM_NTASKS = int(os.environ.get('SLURM_NTASKS', 1))
+
+    # print('maybe_warn_if_slurm SLURM_NTASKS', SLURM_NTASKS)
```
Comment (thequilo) [inline dlp_mpi/util.py]: Debug print?

File: dlp_mpi/util.py [hunk 3]
```diff
@@ -70,3 +70,22 @@ def ensure_single_thread_numeric():
...
+
+    if SLURM_NTASKS > 1:
+        from mpi4py import MPI
+        SIZE = MPI.COMM_WORLD.size
+        # print('maybe_warn_if_slurm SIZE', SIZE)
```
Comment (thequilo) [inline dlp_mpi/util.py]: Debug print?

File: dlp_mpi/mpi.py (hunk 1 above)
Comment (boeddeker) [reply dlp_mpi/mpi.py:90]: Yes, I found it cleaner, with an if statement

File: dlp_mpi/util.py (hunk 2 above)
Comment (boeddeker) [reply dlp_mpi/util.py]: removed

File: dlp_mpi/util.py (hunk 3 above)
Comment (boeddeker) [reply dlp_mpi/util.py]: removed

**Review by thequilo — APPROVED** [review]

**Review by TCord — APPROVED** [review]

**Pull request merged**